### PR TITLE
Track wlroots again

### PR DESCRIPTION
**Pull request by unrelentingtech**
#1351 and more (https://github.com/swaywm/sway/commit/5865af75cf8029cc703cda36b68daafcb658c97b)

hopefully the comment is correct :D

### CODE REVIEW

**Comment by unrelentingtech** [comment]
huh uncrustify just.. crashed?? in CI

File: src/main.cpp [hunk 1]
```diff
@@ -298,7 +298,9 @@ int main(int argc, char *argv[])
     core.display  = display;
     core.ev_loop  = wl_display_get_event_loop(core.display);
     core.backend  = wlr_backend_autocreate(core.display);
-    core.renderer = wlr_backend_get_renderer(core.backend);
+    core.renderer = wlr_renderer_autocreate(core.backend);
```
Comment (ammen99) [inline src/main.cpp:301]: Can we directly create a GLES renderer here? That's the only thing we support.

Comment (unrelentingtech) [reply src/main.cpp:301]: Looks like it is possible, `wlr_gles2_renderer_create_with_drm_fd` is public, I'll try that now

**Comment by droc12345** [comment]
Needed to apply these to the latest wlroots git. Wouldn't compile without them.

Seems to be working fine, as I'm running it now.

**Comment by jbeich** [comment]
After [wlroots@1d3dd7f](https://gitlab.freedesktop.org/wlroots/wlroots/-/commit/1d3dd7fc08f2) this is no longer enough:
```c++
In file included from src/core/core.cpp:17:
src/api/wayfire/nonstd/wlroots-full.hpp:85:10: fatal error: 'wlr/backend/noop.h' file not found
#include <wlr/backend/noop.h>
         ^~~~~~~~~~~~~~~~~~~~
```

**Comment by ammen99** [comment]
To avoid many changes being in the same PR, I think we can merge this and then use another PR for dropping the noop output as well.

@unrelentingtech can you format with uncrustify? (or rebase on top of master, then you'll get a diff in the github actions log)

**Comment by unrelentingtech** [comment]
Now instead of uncrustify crashing, the diff command crashes in CI o_0

**Comment by ammen99** [comment]
> Now instead of uncrustify crashing, the diff command crashes in CI o_0

It doesn't crash, it returns 1 because the code is not formatted :)
If you look at the previous command, it shows the diff you can apply.

**Comment by jbeich** [comment]
> To avoid many changes being in the same PR, I think we can merge this and then use another PR for dropping the noop output as well.

noop fix is probably very simple e.g., https://github.com/freebsd/freebsd-ports/blob/d0c7d052f319/x11-wm/wayfire/files/patch-wlroots-0.15

**Comment by ammen99** [comment]
> noop fix is probably very simple e.g., https://github.com/freebsd/freebsd-ports/blob/d0c7d052f319/x11-wm/wayfire/files/patch-wlroots-0.15

This fix is however not correct. Headless backend has also other uses than what the noop output had. For example, it is perfectly fine to render on a headless backend, if it is used for wayvnc or similar.

**Comment by unrelentingtech** [comment]
well, "other uses" aren't really part of the *fix*, they're extra possibilities for feature development

**Comment by CrezyDud** [comment]
when this

**Comment by ammen99** [comment]
> when this

I just squashed two of the commits and then we can merge the PR. Do note that there are other changes necessary to support wlroots master, so even after merging this PR there is still work to do.

**Comment by ammen99** [comment]
Thanks @unrelentingtech :)

**Comment by CrezyDud** [comment]
> > when this
> 
> I just squashed two of the commits and then we can merge the PR. Do note that there are other changes necessary to support wlroots master, so even after merging this PR there is still work to do.

oof

**Pull request merged**